concepts/05 UI Components/DataGrid/30 Filtering and Searching/6 API/5 Filtering Columns Bound to Collection/20 Server-Side Filtering.md

```diff
@@ -0,0 +1,163 @@
+            >
+            </DxColumn>
+        </DxDataGrid>
+    </template>
+
+    <script>
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DxDataGrid, {
+        DxColumn,
+        DxRemoteOperations,
+    } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxColumn,
+            DxRemoteOperations,
+        },
+        data() {
+            // ...
+        },
+        methods: {
+            calculateFilterExpression(filterValue, selectedFilterOperation, target) {
+                return [
+                    [new DevExpress.data.EdmLiteral("Products/any(o: substringof('" + filterValue + "',o/ProductName) eq true)"), "=", true]
+                ];
+            },
+            calculateCellValue(data) {
+                var res = [];
+                for (var i = 0; i < data.Products.length; i++)
+                    res.push(data.Products[i].ProductName);
+                return res.join(', ');
+            }
+        }
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DataGrid, {
+        Column, 
+    } from 'devextreme-react/data-grid';
+
+    const calculateFilterExpression = (filterValue, selectedFilterOperation, target) => {
+        return [
+            [new DevExpress.data.EdmLiteral("Products/any(o: substringof('" + filterValue + "',o/ProductName) eq true)"), "=", true]
+        ];
+    }
+
+    const calculateCellValue = (data) => {
+        var res = [];
+        for (var i = 0; i < data.Products.length; i++)
+            res.push(data.Products[i].ProductName);
+        return res.join(', ');
+    }
+
+    export default function App() { 
+        return (
+            <DataGrid ... >
+                // ...
+                <Column
+                    dataField="Products"
+                    dataType="string"
+                    calculateFilterExpression={calculateFilterExpression}
+                    calculateCellValue={calculateCellValue}
+                />
+            </DataGrid>
+        );
+    }
+
+---
+
+You need to specify the [calculateFilterExpression](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateFilterExpression) property and put the `"Products/any(o: substringof("filterValue",o/ProductName) eq true)"` expression to the [EdmLiteral](/Documentation/ApiReference/Data_Layer/EdmLiteral/) method. Also, handle the [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue) property and create a comma-separated string from ProductNames. This string will be displayed in column cells.
+
+The DataGrid sends a filter expression to the backend and expects relevant data as a result. The solution for filtering depends on the backend type. However, a common requirement is that the backend should supply the corresponding filtering operation.
```

Comment:
> However, a common requirement is that the backend should **supply** the corresponding filtering operation.

Could you please clarify what do you mean by "supply" here? Right now, "supply" looks like a cheap replacement for "provide".

That's my current take on this sentence. Feel free to clarify should I misinterpret something...

```suggestion
The DataGrid sends a filter expression to the backend and expects relevant data as a result. The solution for filtering depends on the backend type. However, a common requirement is that the backend should implement the corresponding filtering operation.
```

Reply:
In this case, "supply" is equal to "support", but using "implement" is also correct.

Reply:
OK, I see.